### PR TITLE
Display correct build number on download without choosing specific candidate build (#232)

### PR DESCRIPTION
**Pull request by Nebelhom**
Addresses issue #232 

Turned out the buildnumber displayed in downloads was always one too low.

### CODE REVIEW

**Comment by Nebelhom** [comment]
@whimboo 

Please review :)

File: mozdownload/scraper.py [hunk 1]
```diff
@@ -639,7 +639,9 @@ def get_build_info(self):
             self.build_index = 0
             self.logger.info('Selected build: build%s' % self.build_number)
         else:
-            self.logger.info('Selected build: build%s' % self.build_index)
+            # Otherwise displays incorrect build number
```
Comment (whimboo) [inline mozdownload/scraper.py]: Not sure what I should get from this comment. :) I think we can remove it given that your fix is below.

File: mozdownload/scraper.py [hunk 2]
```diff
@@ -639,7 +639,9 @@ def get_build_info(self):
...
+            display_index = str(int(self.build_index) + 1)
+            self.logger.info('Selected build: build%s' % display_index)
```
Comment (whimboo) [inline mozdownload/scraper.py]: Isn't build_index always a number? Also the formatting can automatically convert it to a string. Just pass in the `(index + 1)` in brackets.

Comment (Nebelhom) [inline mozdownload/scraper.py]: *sigh* Another one of those school boy errors of mine. @whimboo as an explanation. The following happened. I wrote the line

>self.logger.info('Selected build: build%d' % self.build_index + 1)

Which threw the following error:

>  INFO | Retrieving list of candidate builds from https://ftp.mozilla.org/pub/mozilla.org/firefox/candidates/36.0-candidates/
  INFO | Found 2 builds: build1, build2
Traceback (most recent call last):
  File "./scraper.py", line 1243, in <module>
    cli()
  File "./scraper.py", line 1235, in cli
    build = BUILD_TYPES[options.type](**kwargs)
  File "./scraper.py", line 628, in __init__
    Scraper.__init__(self, *args, **kwargs)
  File "./scraper.py", line 144, in __init__
    self.get_build_info()
  File "./scraper.py", line 643, in get_build_info
    self.logger.info('Selected build: build%d' % self.build_index + 1)
TypeError: cannot concatenate 'str' and 'int' objects

Consequently, I assumed that `self.build_index` is a string.

What do we learn from this? kids, don't code when you are tired ;)

**Comment by Nebelhom** [comment]
Last commit failed because the line was one symbol too long :(

**Comment by Nebelhom** [comment]
@whimboo This should be all in order now. Could you please review it. Thank you :)

**Comment by Nebelhom** [comment]
*sigh* I don't know what is going on here. The two failures that Travis is showing me here, do not show up on my system and I have the pep8 hooks active. 

In fact, it showed me this exact error before and I fixed it.

fyi. error is 'mozdownload/scraper.py:643:29: E128 continuation line under-indented for visual indent'

Any suggestions are welcome. I will revisit at another time. It is probably something simple that I am missing now...

**Comment by parkouss** [comment]
Hello Nebelhom,
Since travis always install new tools, that must indicate that your version of pep8 is not the latest.
Try "pip install -U pep8" and re run your local command. :)

**Comment by parkouss** [comment]
Hum sorry for my last comment I just realized that it is probably not that - sorry for the noise;;

**Comment by Nebelhom** [comment]
@parkouss you put me on the right track, thanks :) It was simple, just as I thought ;)

@whimboo Should now really be ready for review... Thanks :)

**Comment by whimboo** [comment]
I think we should allow longer lines for Python code. Somewhat like 99 chars instead of 79. This makes us more flexible and doesn't blight the code that much. Feel free to raise a follow-up bug. Maybe we can do it similar to my request on [bug 1138481](https://bugzilla.mozilla.org/show_bug.cgi?id=1138481)

**Comment by whimboo** [comment]
I merged this PR as https://github.com/mozilla/mozdownload/commit/7192df5b81fc92b044160769f3624454245d36a1. Thank you @Nebelhom!

**Pull request closed without merging**